src/nvim/mbyte.c

```diff
@@ -722,6 +788,15 @@ bool utf_composinglike(const char *p1, const char *p2)
   return arabic_combine(utf_ptr2char(p1), c2);
 }
 
+/// Check if the next character is a composing character when it
+/// comes after the first. For Arabic sometimes "ab" is replaced with "c", which
+/// behaves like a composing character.
+bool composinglike(int32_t const first, int32_t const next)
```

Comment:
This name is strange without a `utf_` prefix. Maybe name it `utf_char_composinglike()`?